### DISCUSSION TITLE
How to add parameters to With EventSource

### DISCUSSION THREAD

**Discussion by zahrat** [enisdenjo/graphql-sse#73] [Q&A]
I want to use graphql-sse in client-side ( with react-native).  I read this [link](https://the-guild.dev/graphql/sse/recipes) but How to pass parameters to query? for example for subscription I have to pass userId , it's something like this:
`  subscription ($userId: Int!) {
    notificationAdded(userId: $userId) {
      text
      isReaded
      notificationType
      senderId
      id
    }
  }`
  how to do that? I try this but its not working:
  `const url = new URL(
      api_url,
    );
    url.searchParams.append(
      'query',
      notificationAddedSubscription
    );
    url.searchParams.append('variables', JSON.stringify({userId: 5}));`

**Comment by enisdenjo** [comment]
That is exactly how you should pass the variables. Here's the code excerpt in graphql-sse which does the parsing of variables in the URL:

https://github.com/enisdenjo/graphql-sse/blob/47ea198dbb811192784c7bb600fcc3cdcedb7538/src/handler.ts#L1021-L1022

Are you using the graphql-sse server? If not, you should open an issue on the relevant repo.

**Comment by zahrat** [comment]
Appending query&variables to url in react native not working for me. I passed them in body of request and it's worked.
